lib/LocalVerifiableCredentialStore.js

```diff
@@ -208,31 +209,18 @@ export class LocalVerifiableCredentialStore {
     const query = Array.isArray(credentialQuery) ? credentialQuery :
       [credentialQuery];
 
-    const _query = async ({example}) => {
-      const {type, trustedIssuer = []} = example;
+    const _query = async ({example, trustedIssuer = []}) => {
+      const {type} = example;
       // normalize trusted issuers to be an array
       const trustedIssuers = Array.isArray(trustedIssuer) ? trustedIssuer :
         [trustedIssuer];
 
-      // build query to find all VCs that match any combination of type+issuer
-      const query = [];
-      const issuers = trustedIssuers.filter(({required}) => required)
-        .map(({issuer}) => issuer);
+      const issuers = trustedIssuers.map(({issuer}) => issuer);
       const types = Array.isArray(type) ? type : [type];
-      for(const type of types) {
-        if(issuers.length === 0) {
-          query.push({type});
-          continue;
-        }
-        for(const issuer of issuers) {
-          query.push({type, issuer});
-        }
-      }
-      const queryType = query.map(q => q.type);
-      const queryTypes = Array.isArray(queryType) ? queryType : [queryType];
       const selector = {
         'meta.profileId': profileId,
-        'content.type': {$in: queryTypes}
+        'content.type': {$in: types},
+        $or: [{'meta.issuer': {$in: issuers}}]
```

Comment:
So we want the `$or` conditional to do this:

1. Find all documents with `meta.profileId == X` AND `content.type in Y` AND `meta.issuer in Z1`.
... OR ...
2. Find all documents with `meta.profileId == X` AND `content.type in Y` AND `meta.issuer in Z2`.
... OR ...
3. Find all documents with `meta.profileId == X` AND `content.type in Y` AND `meta.issuer in Z3`.

The `$or` you've built here only has one option in it. I expect that you've got have just `$or` at the top level of the query and then have a full query for every issuer that goes underneath it. Something that looks a bit like this:

```js
const $or = issuers.map(issuer => ({
  'meta.profileId': profileId,
  'content.type': {$in: types},
  'meta.issuer': issuer
}));

const selector = {$or};
```

But you'll have to deal with optionality around whether any `issuers` were specified.

Reply:
You may be able to alternatively just put `meta.profileId` and `content.type` at the top level and have `$or` be filled with an array of `{'meta.issuer': issuer1}, {'meta.issuer': issuer2}, ...{'meta.issuer': issuerN}`. But you don't want to use `$in` here, because you want separate results for each issuer.